features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -114,6 +114,9 @@ void ATHandler::set_debug(bool debug_on)
 
 ATHandler::~ATHandler()
 {
+    set_is_filehandle_usable(false);
+    set_file_handle(NULL);
```

Comment:
That's not legal - you call `set_blocking` on NULL. 

Thought `set_is_filehandle_usable(false)` should do it. Wasn't that what you used to call to attach/detach sigio around PPP? Although that seems gone.